packages/flutter_tools/lib/src/test/flutter_tester_device.dart

```diff
@@ -261,6 +268,29 @@ class FlutterTesterTestDevice extends TestDevice {
     );
   }
 
+  Future<void> _startDevTools(Uri forwardingUri, DartDevelopmentService? dds) async {
```

Comment:
Do we already have code similar to this anywhere else in the tool?

Reply:
We do have code that's tightly coupled to `ResidentRunner` in the form of `ResidentDevtoolsHandler`. I pulled the relevant code out for use here since `flutter test` doesn't require any of the functionality from `ResidentDevtoolsHandler`.